src/main/kotlin/FiToBookConvertor.kt

```diff
@@ -11,76 +11,112 @@ import java.time.format.DateTimeFormatter
+        summRow: Int? = null,
+        summAmountCol: String? = null,
+        summBalCol: String? = null
+    ) {
+        val trades = activities.filter { it.type in TRADE_TYPES }
+        val toClipboard = buildString { // Inline string building: Used buildString for better memory allocation
+            when (mode) {
+                "0" -> processForExeTab(trades, startingRow)
+                "1" -> processForSummary(activities, startingRow, trades.size, summRow, summAmountCol, summBalCol)
+            }
+            // Remove trailing newline
+            if (isNotEmpty() && last() == NEWLINE) deleteCharAt(length - 1)
+        }
+
+        println("New clipboard:")
+        println(toClipboard)
+        writeToClipboard(toClipboard)
+        printDiscrepancies(trades)
+    }
+
+    private fun StringBuilder.processForExeTab(trades: List<Activity>, startingRow: Int) {
+        trades.forEachIndexed { index, trade ->
+            append(formatExecutionRow(trade, startingRow + index))
+            append(NEWLINE)
+        }
+    }
+
+    private fun StringBuilder.processForSummary(
+        activities: List<Activity>,
+        startingRow: Int,
+        tradesSize: Int,
         summRow: Int?,
         summAmountCol: String?,
         summBalCol: String?
     ) {
-        val dateFormatter = DateTimeFormatter.ofPattern("MM/dd/yyyy")
-        val toClipboard = StringBuilder()
-        val trades = activities.filter { it.type in setOf(ActivityType.BOUGHT, ActivityType.SOLD) }
-
-        when (mode) {
-            "0" -> processForExeTab(trades, dateFormatter, startingRow, toClipboard)
-            "1" -> processForSummary(
-                activities = activities,
-                dateFormat = dateFormatter,
-                startingRow = startingRow,
-                tradesSize = trades.size,
-                toClipboard = toClipboard,
-                summTabStartingRow = summRow,
-                summTabAmountCol = summAmountCol,
-                summTabBalCol = summBalCol
-            )
-        }
+        var row = startingRow + tradesSize - 1
+        var currentSummRow = summRow
+
+        activities.asReversed().forEach { activity ->
+            when (activity.type) {
+                in TRADE_TYPES -> append(formatSummaryTrade(row--, LocalDate.now().year))
+                else -> append(formatSummaryNonTrade(activity))
+            }
 
-        // Remove the last newline character
-        if (toClipboard.isNotEmpty()) {
-            toClipboard.deleteCharAt(toClipboard.length - 1)
+            if (summRow != null && summAmountCol != null && summBalCol != null) {
+                append(TAB).append("=$summBalCol$currentSummRow+$summAmountCol${++currentSummRow}")
+            }
+            append(NEWLINE)
         }
+    }
 
-        println("New clipboard:")
-        println(toClipboard)
-        writeToClipboard(toClipboard.toString())
+    private fun formatExecutionRow(activity: Activity, row: Int): String {
```

Comment:
```suggestion
    private fun toInvestBookExecutionFromActivity(activity: Activity, row: Int): String {
```
keep original method name